### ISSUE TITLE
[🐛] config IDs when using app.config.js/app.config.ts rather than app.json

### ISSUE THREAD

**Issue by megacherry** [invertase/react-native-google-mobile-ads#462]
### What happened?

First of all: sorry if any information or similar is missing. I try my best :)

I'm writing because I currently switched from app.json to app.config.js/app.config.ts (I use expo). Mainly to serve different environments.

Then when I went to rebuild my project, I got an error: "error: ios_app_id key not found in react-native-google-mobile-ads key in app.json. App will crash without it."

Then I searched for related issues and found the following issue: https://github.com/invertase/react-native-google-mobile-ads/issues/370

But that doesn't help me because firstly the issue is closed and secondly the pull request is only for Android (and not merged). I currently have the problems on iOS.

I then looked at what it fails and found that a parsing of Javascript Object Literal Syntax is not so trivial. Especially something like this: "process.env.APP_NAME || 'myapp'".

So I came up with the following solution (ios_config.sh):
- Search additionally for app.config.js/app.config.ts.
- If found, then try to parse an additional file "admob-config.json" on the corresponding keys

I patched the ios_config.sh locally. But that would be only the part for iOS. If the solution sounds good, then we have to adjust the readme and think of something for Android as well.

How can I best help you at this point?

### Platforms

Android and iOS

### React Native Info

```Shell
System:
  OS: macOS 13.6
  CPU: (8) arm64 Apple M1
  Memory: 91.02 MB / 16.00 GB
  Shell:
    version: 3.2.57
    path: /bin/bash
Binaries:
  Node:
    version: 20.7.0
    path: /opt/homebrew/bin/node
  Yarn:
    version: 3.6.1
    path: /opt/homebrew/bin/yarn
  npm:
    version: 10.1.0
    path: /opt/homebrew/bin/npm
  Watchman:
    version: 2023.09.25.00
    path: /opt/homebrew/bin/watchman
Managers:
  CocoaPods:
    version: 1.12.0
    path: /Users/andreas.kirsch/.rbenv/shims/pod
SDKs:
  iOS SDK:
    Platforms:
      - DriverKit 23.0
      - iOS 17.0
      - macOS 14.0
      - tvOS 17.0
      - watchOS 10.0
  Android SDK:
    API Levels:
      - "28"
      - "30"
      - "31"
      - "32"
      - "33"
    Build Tools:
      - 29.0.2
      - 30.0.2
      - 30.0.3
      - 31.0.0
      - 33.0.0
    System Images:
      - android-30 | Google Play ARM 64 v8a
      - android-31 | Google APIs ARM 64 v8a
      - android-33 | Google APIs ARM 64 v8a
    Android NDK: 22.1.7171670
IDEs:
  Android Studio: 2022.1 AI-221.6008.13.2211.9514443
  Xcode:
    version: 15.0/15A240d
    path: /usr/bin/xcodebuild
Languages:
  Java:
    version: 17.0.8.1
    path: /opt/homebrew/bin/javac
  Ruby:
    version: 3.0.2
    path: /Users/andreas.kirsch/.rbenv/shims/ruby
npmPackages:
  "@react-native-community/cli": Not Found
  react: Not Found
  react-native: Not Found
  react-native-macos: Not Found
npmGlobalPackages:
  "*react-native*": Not Found
Android:
  hermesEnabled: Not found
  newArchEnabled: Not found
iOS:
  hermesEnabled: Not found
  newArchEnabled: Not found
```

### Are your using Typescript?

- [X] My project is using Typescript

### package.json

```JSON
not relevant at this point
```

### app.json

```JSON
not relevant at this point

the template title should be changed to "app.json/app.config.js/app.config.ts" :)
```

**Comment by robgraeber** [comment]
+1

**Comment by robgraeber** [comment]
.

**Comment by MorvanL** [comment]
+1

**Comment by manarbenkraouda** [comment]
+1

**Comment by mikehardy** [comment]
I believe reasonable PRs are merged here - if you post a PR we can discuss, and there seems to be a lot of interest from the `+1` brigade posting copycat comments, so perhaps we'll have people willing to do more than post a +1 and they'll test it and give us confidence it works prior to merge...

**Comment by MorvanL** [comment]
Hi @mikehardy. 
I'm looking for a solution. On my side, the problem comes from the Ruby command [in this line](https://github.com/invertase/react-native-google-mobile-ads/blob/ed945edd467268fc25af1b5387b2a9aa2d5a95c7/ios_config.sh#L80) (same on L44 too). The code seems fine, but it looks like I need to escape the quotes for the `JSON.parse` to work. 
The string representing the contents of the app.json file is passed to `json.parse(app_json_str)` but it only works if the `"` are replaced by `\"` in the string. I can't figure out why this happens, as the ruby syntax seems fine. 
If this problem is real, then everyone should have a problem building on iOS. Why are we so few ??? I haven't found the answer yet.

**Comment by mikehardy** [comment]
Perhaps it's the old system ruby vs a modern one? Or old system python vs modern one?

**Comment by mikehardy** [comment]
That seems unrelated to the original issue from original poster, worth noting...

**Comment by MorvanL** [comment]
I use python 3.9.13 and I tested with ruby 3.2.2 and ruby 2.6.10 so I don't think. I'm on a zsh bash (i don't know if it coulb be the reason) and on a intel mac (2020) macOS Ventura 13.4.1. 

I have exactly the same error message as the author, hence my initial +1. I wonder, though, if the author didn't actually have the same problem as me. 

I'll try building on the EAS server to see if the error also occurs.

**Comment by MorvanL** [comment]
Finally my problem is caused by a `'` in a value of my app.json config which crashes ruby's json parser which interprets the `'` character.
I apologize for the inconvenience...

**Comment by megacherry** [comment]
hi @mikehardy, thanks for your answer. but unfortunately I cannot push PRs. iOS is fixed, have to write documentation and for android i need help from other supporters.

**Comment by megacherry** [comment]
@mikehardy: The PR is created. Took me some time because it's my first PR on GitHub. :)

**Comment by dylancom** [comment]
Solved by #517
